boskos/cmd/cli/cli.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	"k8s.io/test-infra/boskos/client"
+	"k8s.io/test-infra/boskos/common"
+)
+
+type options struct {
+	// common, used to create the client
+	serverURL string
+	ownerName string
+
+	c *client.Client
+
+	acquire acquireOptions
+}
+
+type acquireOptions struct {
+	requestedType  string
+	requestedState string
+	targetState    string
+	wait           bool
+}
+
+func main() {
+	options := options{}
+
+	root := &cobra.Command{
+		Use:   "boskosctl",
+		Short: "Boskos command-line client for resource leasing",
+		Long: `Boskos provides a flexible resource leasing server.
+
+The boskosctl is a command-line client for this server,
+allowing for a user to acquire and release leases from
+scripts with a simple interface.`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// the root command does nothing, so just print help
+			return cmd.Help()
+		},
+		Args: cobra.NoArgs,
+	}
+	root.PersistentFlags().StringVar(&options.serverURL, "server-url", "", "URL of the Boskos server")
+	root.PersistentFlags().StringVar(&options.ownerName, "owner-name", "", "Name identifying the user of this client")
+	for _, flag := range []string{"server-url", "owner-name"} {
+		if err := root.MarkPersistentFlagRequired(flag); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+	}
+
+	acquire := &cobra.Command{
```

Comment:
maybe acquire should be in its own file, at least its own function. We probably don't want to cluster everything in main.

Reply:
//TODO